libsession/src/main/java/org/session/libsession/snode/OnionRequestAPI.kt

```diff
@@ -190,8 +191,19 @@ object OnionRequestAPI {
                 if (unusedSnodes.count() < pathSnodeCount) { throw InsufficientSnodesException() }
                 // Don't test path snodes as this would reveal the user's IP to them
                 guardSnodes.minus(reusableGuardSnodes).map { guardSnode ->
-                    val result = listOf( guardSnode ) + (0 until (pathSize - 1)).map {
-                        val pathSnode = unusedSnodes.getRandomElement()
+                    val result = listOf( guardSnode ) + (0 until (pathSize - 1)).mapIndexed() { index, _ ->
+                        var pathSnode = unusedSnodes.getRandomElement()
+
+                        // we want to make sure the last node in the path is above version 2.8.0
+                        // to help with an issue that will disappear once the nodes are all updated
+                        if(index == pathSize - 2) {
+                            // because we are now grabbing the whole node pool there should always
+                            // be a node that is above version 2.8.0
+                            while(Util.compareVersions(pathSnode.version, "2.8.0") < 0) {
+                                pathSnode = unusedSnodes.getRandomElement()
+                            }
```

Comment:
Is the assumption "should always be a node that is above version 2.8.0" reliable? I'm a little bit worried that this can create a infinite loop in exceptional cases, if the server side ends up breaking the assumption. Maybe remove the node from the comparison pool and check if the pool still has something as a safe guard?

Reply:
This may be one of the implementations?
```suggestion
  pathSnode = unusedSnodes.shuffled().firstOrNull { Util.compareVersions(pathSnode.version, "2.8.0") >= 0 }
```

Reply:
I was wondering the same but Session controls about 10% of the nodes and they are guaranteed to be updated.
This patch will also be removed on the next release as all nodes will have been updated by then.